### ISSUE TITLE
`EnterpriseSearchRestIT` failure

### ISSUE THREAD

**Issue by eyalkoren** [elastic/elasticsearch#95917]
### Context
With https://github.com/elastic/elasticsearch/pull/95775, enterprise search adds the `behavioral_analytics-events-final_pipeline` lazily only when the data stream is created. This contradicts a validation that is about to be added within https://github.com/elastic/elasticsearch/pull/95782, which does not allow to install an index template if it refers to a non-existing `default_pipeline` or `final_pipeline`.

As agreed with @jimczi, we will disable `EnterpriseSearchRestIT` for now, so that it doesn't block https://github.com/elastic/elasticsearch/pull/95782 and figure out how to handle that later on.

One optional solution we discussed is changing [`behavioral_analytics-events-final_pipeline`](https://github.com/elastic/elasticsearch/blob/f02e9f9c0ad03ff874368cc24f0e287fe43953b2/x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/entsearch/analytics/behavioral_analytics-events-final_pipeline.json) to contain:
```
{
  "processors": [
    {
      "pipeline" : {
        "name": "lazily-added",
        "ignore_missing_pipeline": true,
        "description": "The actual pipeline, which is installed lazily"
      }
    }
  ]
}
```
and install the `lazily-added` upon the proper trigger (data stream creation or other).

Note that while this will prevent dependency-related errors, it may still imply there is an undesirable race condition between the installation of the ingest pipeline and the referring index template.

### CI Link

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+part-2/33583/console

### Repro line

./gradlew ':x-pack:plugin:ent-search:qa:rest:yamlRestTest' --tests "org.elasticsearch.xpack.entsearch.EnterpriseSearchRestIT.test {p0=entsearch/80_behavioral_analytics_delete/Delete Analytics Collection - Analytics Collection does not exist}" -Dtests.seed=714C8C6B612820A4

### Does it reproduce?

Yes

### Applicable branches

eyalkoren:ingest-pipeline-registry

### Failure excerpt

```
"type" : "status_exception",
"reason" : "error while creating analytics collection [my-test-analytics-collection]", 
"caused_by" : { 
"type" : "illegal_argument_exception", 
"reason" : "no matching index template found for data stream [behavioral_analytics-events-my-test-analytics-collection]", 
"stack_trace" : "java.lang.IllegalArgumentException: no matching index template found for data stream [behavioral_analytics-events-my-test-analytics-collection] |  
  | at org.elasticsearch.server@8.9.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataCreateDataStreamService.lookupTemplateForDataStream(MetadataCreateDataStreamService.java:307) |  
  | at org.elasticsearch.server@8.9.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataCreateDataStreamService.createDataStream(MetadataCreateDataStreamService.java:221) |  
  | at org.elasticsearch.server@8.9.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataCreateDataStreamService.createDataStream(MetadataCreateDataStreamService.java:168) |  
...
```

**Comment by eyalkoren** [comment]
@jimczi #95782 has been merged.
As I said offline, with the validation described above, this is more than a test failure. It is a real issue that needs to be addressed.
I am not the one to decide on prioritization, so I can't decide how to label this issue, but I propose to treat it as a `bug` that must be fixed for 8.9.
Thanks for your assistance! 🙏

**Comment by joemcelroy** [comment]
We decided short term that we will incorporate the changes @eyalkoren made and installing the pipeline immediately, just so we can enable the EnterpriseSearchRestIT yaml tests + unblock analytics on main. This has now been merged 🎉 https://github.com/elastic/elasticsearch/pull/96104.

Long term we are going to find a way to lazily load the geoip database when its needed.
